src/nvim/option.c

```diff
@@ -2557,6 +2557,76 @@ static bool valid_spellfile(const char_u *val)
   return true;
 }
 
+/// Handle setting 'mousescroll'.
+/// @return error message, NULL if it's OK.
+static char_u* check_mousescroll(char_u *string) {
+  long vertical = -1;
+  long horizontal = -1;
+
+  for (;;) {
+    char_u *end = vim_strchr(string, ',');
+    size_t length = end ? (size_t)(end - string) : STRLEN(string);
+
+    // Both "ver:" and "hor:" are 4 bytes long.
+    // They should be followed by at least one digit.
+    if (length <= 4) {
+      return e_invarg;
+    }
+
+    long *direction;
+
+    if (memcmp(string, "ver:", 4) == 0) {
+      direction = &vertical;
+    } else if (memcmp(string, "hor:", 4) == 0) {
+      direction = &horizontal;
+    } else {
+      return e_invarg;
+    }
+
+    // If the direction has already been set, this is a duplicate.
+    if (*direction != -1) {
+      return e_invarg;
+    }
+
+    // Verify that only digits follow the colon.
+    for (size_t i=4; i<length; ++i) {
+      if (!ascii_isdigit(string[i])) {
+        return (char_u *)N_("E548: digit expected");
+      }
+    }
+
+    *direction = strtol((char*)string + 4, NULL, 10);
+
+    // Num options are generally kept within the signed int range.
+    // We know this number won't be negative because we've already checked for
+    // a minus sign. We'll allow 0 as a means of disabling mouse scrolling.
+    if (*direction > INT_MAX) {
+      return e_invarg;
+    }
+
+    if (!end) {
+      break;
+    }
+
+    string = end + 1;
+  }
+
+  // If a direction wasn't set, fallback to the default value.
+  if (vertical == -1) {
+    p_mousescroll_vert = MOUSESCROLL_VERT_DFLT;
+  } else {
+    p_mousescroll_vert = vertical;
+  }
+
+  if (horizontal == -1) {
+    p_mousescroll_hor = MOUSESCROLL_HOR_DFLT;
+  } else {
+    p_mousescroll_hor = horizontal;
```

Comment:
nit: prefer ternaries for these simple if-else cases, it results in much shorter code without a loss of clarity.

Reply:
Valid point, I made the change 👍